libraries/AP_RangeFinder/AP_RangeFinder_RDS02UF.cpp

```diff
@@ -0,0 +1,222 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * RDS02UF Note:
+ * Sensor range scope 1.5m~20.0m
+ * Azimuth Coverage ±17°,Elevation Coverage ±3°
+ * Frame Rate 20Hz
+ */
+#include "AP_RangeFinder_RDS02UF.h"
+
+#include <AP_HAL/AP_HAL.h>
+#include <ctype.h>
+
+#define RDS02_HEAD1 0x55
+#define RDS02_HEAD2 0x55
+#define RDS02_END 0xAA
+
+#define RDS02UF_HEAD_LEN 2
+#define RDS02UF_PRE_DATA_LEN 6
+#define RDS02UF_DATA_LEN 10
+#define RDS02_DATA_Y_INDEX_L 13
+#define RDS02_DATA_Y_INDEX_H 14
+#define RDS02_DATA_FC_L    8
+#define RDS02_DATA_FC_H    9
+#define RDS02_TARGET_INFO  0x70C
+#define RDS02UF_DIST_MAX_CM           2000
+#define RDS02UF_DIST_MIN_CM           150
+#define AP_RANGEFINDER_RDS02UF_TIMEOUT_MS   200
+
+
+extern const AP_HAL::HAL& hal;
+
+AP_RangeFinder_RDS02UF::AP_RangeFinder_RDS02UF(
+    RangeFinder::RangeFinder_State &_state,
+    AP_RangeFinder_Params &_params):
+    AP_RangeFinder_Backend_Serial(_state, _params)
+{
+    
+}
+
+// return last value measured by sensor
+bool AP_RangeFinder_RDS02UF::get_reading(float &reading_m)
+{
+    if (uart == nullptr) {
+        return false;
+    }
+
+    // read any available lines from the lidar
+    float sum = 0.0f;
+    uint16_t count = 0;
+    int16_t nbytes = uart->available();
+    while (nbytes-- > 0) {
+        uint8_t c = uart->read();
+        if (decode(c)) {
+            sum += _distance_m;
+            count++;
+        }
+    }
+
+    // return false on failure
+    if (count == 0) {
+        return false;
+    }
+
+    if (AP_HAL::millis() - state.last_reading_ms > AP_RANGEFINDER_RDS02UF_TIMEOUT_MS) {
+        set_status(RangeFinder::Status::NoData);
```

Comment:
We will never get here if the device simply stops supplying data as you're returning early in that case.

Perhaps move this check into the early-return block.

Structure-wise this is odd as the timestamp is set elsewhere.  We should move this line up into here when we do get a non-zero count:
```
                        state.last_reading_ms = AP_HAL::millis();
```